python/grep.py

```diff
@@ -1009,14 +1012,17 @@ def grep_process(*args):
     except Exception as e:
         result = e
 
-    return pickle.dumps(result)
-
-grep_stdout = grep_stderr = ''
+    return pickle.dumps(result, 0)
 
 def grep_process_cb(data, command, return_code, out, err):
     global grep_stdout, grep_stderr, matched_lines, hook_file_grep
 
+    if isinstance(out, str):
+        out = out.encode()
```

Comment:
I think you should do that only in Python 3 (same comment for the second encode below).
With Python 2, the type is always str and if there are unicode chars, this will raise a UnicodeDecodeError.

Reply:
By the way, Python 2 support is now optional, so we might drop Python 2 support now.
But since a lof of people is using the script, we could keep the compatibility for some time.

Reply:
Nah, I did test on python2 (well just the function, not whole script) -- string and byte alike will both match the isinstance but the .encode() will be no-op here as pickle with version 0 guarantees pure ascii. It should work.

I don't have any recent weechat with python2 to test... I can try on a way older weechat probably let me see

Reply:
confirmed, it works fine on weechat 2.3 with python2.

Reply:
Did you try with accents?

For example:

    Python 2.7.18 (default, Apr 20 2020, 20:30:41) 
    [GCC 9.3.0] on linux2
    Type "help", "copyright", "credits" or "license" for more information.
    >>> 'sébastien'
    's\xc3\xa9bastien'
    >>> 'sébastien'.encode()
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 1: ordinal not in range(128)

Reply:
There are weird characters in the log file I tried, but that doesn't matter - the only lines going in through that callback come from `pickle.dumps(result, 0)` which is guaranteed to be ascii. If that ever returns non-ascii then the python3 version won't work; I couldn't get the default pickle which isn't ascii (outputs invalid utf8) to work.

If you really don't want to call it in python2 I don't mind but I have no idea how to check current python version so you'll need to spoonfeed me that, I have no interest in spending the time to figure more python2/3 compat things out..